client/components/data/query-simple-payments/index.jsx

```diff
@@ -0,0 +1,60 @@
+/**
+ * External dependencies
+ */
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { requestProduct, requestProducts } from 'state/simple-payments/product-list/actions';
+
+class QuerySimplePayments extends Component {
+	componentWillMount() {
+		this.request( this.props );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if (
+			nextProps.siteId !== this.props.siteId ||
+			nextProps.productId !== this.props.productId
+		) {
+			this.request( nextProps );
+		}
+	}
+
+	request( props ) {
+		const { siteId } = props;
+
+		if ( ! props.siteId ) {
+			return;
+		}
+
+		if ( props.allProducts ) {
+			props.requestProducts( siteId );
+		}
+
+		if ( props.productId ) {
+			props.requestProduct( siteId, props.productId );
+		}
+	}
+
+	render() {
+		return null;
+	}
+}
+
+QuerySimplePayments.propTypes = {
+	allProducts: PropTypes.bool,
+	siteId: PropTypes.number.isRequired,
+	requestProduct: PropTypes.func,
+	requestProducts: PropTypes.func,
+};
+
+QuerySimplePayments.defaultProps = {
+	allProducts: false,
+	requestProduct: () => {},
+	requestProducts: () => {},
```

Comment:
If a component is immediately `connect`ed and only then exported, I wouldn't bother with declaring `requestProduct` and `requestProducts` in `propTypes`. It's just an additional verbosity, as it's obvious that the props are always present. What about declaring just the props that are part of the component's public API?